### PR TITLE
[Experiment] Enable GitHub Actions

### LINKED ISSUE #227: Migrate to GH Actions
We're still on Travis CI here. I'm not entirely sure how this should be migrated.

But the migration should also update CI to [fail when warnings are emitted](https://github.com/purescript/trypurescript/pull/224#issuecomment-850716586)

### PR DESCRIPTION
**Pull request by thomashoneyman**
**Description of the change**

Travis.org no longer runs CI jobs, which means the CI and release scripts for this repository will no longer run at all. This PR is an experiment to switch to GitHub Actions instead. Closes #227.

---

**Checklist:**

- [x] Added the change to the changelog's "Unreleased" section with a reference to this PR (e.g. "- Made a change (#0 by @)")
- [x] Linked any existing issues or proposals that this pull request should close
- [ ] Updated or added relevant documentation
- [x] Added a test for the contribution (if applicable)

### CODE REVIEW

**Comment by thomashoneyman** [comment]
I'm not totally sure how to properly set up caching for the Stack build, but it should be doable with the cache action.

**Comment by thomashoneyman** [comment]
OK -- this looks to be working all the way from caching to automating artifact uploads on release. I can't totally check the last step until we actually make a release, but I _think_ it should work.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+          path: |
+            ~/.stack
+            .stack-work
+          key: ${{ runner.os }}-stack-${{ hashFiles('stack.yaml.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-stack
+
+      - name: Build server code
+        run: |
+          timeout 45m stack --no-terminal -j1 build
```
Comment (hdgarrood) [inline .github/workflows/ci.yml:34]: GitHub actions’ time limits are much more generous than Travis’, so I think we can probably remove the `timeout`

**Review by hdgarrood — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+jobs:
+  build_server:
+    name: Build server
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: haskell/actions/setup@v1
+        with:
+          enable-stack: true
```
Comment (JordanMartinez) [inline .github/workflows/ci.yml:20]: Should we pin the stack version here at all? See for example, the [PureScript ci.yml file](https://github.com/purescript/purescript/blob/master/.github/workflows/ci.yml#L37-L39).

Also, should we set the [`stack-no-global` flag to `true`](https://github.com/haskell/actions/tree/main/setup#inputs)?

**Pull request merged**